dimension/dimension.go

```diff
@@ -83,13 +89,86 @@ func (s *Store) AddNodeID(w http.ResponseWriter, r *http.Request) {
 	value := vars["value"]
 	nodeID := vars["node_id"]
 
-	dim := models.Dimension{Name: dimensionName, Value: value, NodeID: nodeID, InstanceID: id}
+	dim := models.DimensionOption{Name: dimensionName, Value: value, NodeID: nodeID, InstanceID: id}
 	if err := s.UpdateDimensionNodeID(&dim); err != nil {
 		log.Error(err, nil)
 		handleErrorType(err, w)
 	}
 }
 
+func (s *Store) GetDimensions(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	datasetID := vars["id"]
+	editionID := vars["edition"]
+	versionID := vars["version"]
+
+	results, err := s.Storer.GetDimensions(datasetID, editionID, versionID)
+	if err != nil {
+		handleErrorType(err, w)
+		return
+	}
+
+	bytes, err := json.Marshal(results)
+	if err != nil {
+		log.Error(err, log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID})
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeBody(w, bytes)
+	log.Debug("get dimensions", log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID})
+
+}
+
+func (s *Store) GetDimensionOptions(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	datasetID := vars["id"]
+	editionID := vars["edition"]
+	versionID := vars["version"]
+	dimension := vars["dimension"]
+	results, err := s.Storer.GetDimensionOptions(datasetID, editionID, versionID, dimension)
+	if err != nil {
+		handleErrorType(err, w)
+		return
+	}
+
+	bytes, err := json.Marshal(results)
+	if err != nil {
+		log.Error(err, log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID, "dimension": dimension})
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeBody(w, bytes)
+	_, err = w.Write(bytes)
+	if err != nil {
+		log.Error(err, log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID, "dimension": dimension})
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	log.Debug("get dimension options", log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID, "dimension": dimension})
+}
+
+// CreateDataset manages the creation of a dataset from a reader
+func unmarshalDimensionCache(reader io.Reader) (*models.CachedDimensionOption, error) {
+	bytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, errors.New("Failed to read message body")
+	}
+
+	var option models.CachedDimensionOption
+
+	err = json.Unmarshal(bytes, &option)
+	if err != nil {
+		return nil, errors.New("Failed to parse json body")
+
+	}
+	if option.Name == "" || (option.Value == "" && option.CodeList == "") {
+		log.Debug(".....", log.Data{"data": option})
```

Comment:
"...." is probably not a useful debug message?

Reply:
This was a debug message, It is no longer needed. so I have removed it